clients/vscode-hlasmplugin/scripts/syntaxes/tmgrammarGenerator.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2023 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as fs from 'fs'
+
+const syntaxesDir = "syntaxes";
+
+const grammar_details_template = './scripts/syntaxes/hlasm_tmgrammar_template.txt';
+const grammar_base_template = './scripts/syntaxes/hlasm_base_template.txt';
+
+const code_block_listing_begin = '.{2}Loc  Object Code    Addr1 Addr2  Stmt   Source Statement.*';
+const code_block_listing_long_begin = '.{2}Loc    Object Code      Addr1    Addr2    Stmt  Source Statement.*';
+const code_block_listing_annotation_length = '.{40}';
+const code_block_listing_long_annotation_length = '.{49}';
+
+interface GrammarFile {
+  file: string;
+  grammarName: string;
+  scope: string;
+}
+
+interface GrammarDetails extends GrammarFile {
+  entryPattern: string;
+  codeBlockBegin: string;
+  beginLineSkipRule: string;
+  listingOffset: string;
+}
+
+const hlasmGeneralGrammar: GrammarDetails = {
+  file: './syntaxes/hlasmGeneral.tmLanguage.json',
+  grammarName: 'IBM HLASM General',
```

Comment:
I haven't realized what we call the grammar. @zeibura is that an issue?

Reply:
Besides other references to IBM, there's also a mention about ibm assembler in the `package.json` which is reflected on the Marketplace. Not sure if that might be an issue as well:

![image](https://github.com/BroadcomMFD-DevOps/che-che4z-lsp-for-hlasm/assets/99467904/0e5885d6-790d-4c73-8484-1aa736c6d2f2)

![image](https://github.com/BroadcomMFD-DevOps/che-che4z-lsp-for-hlasm/assets/99467904/456d87c6-d889-4ecf-8147-07bbb86b5043)

Reply:
About the keywords - It's not coming from `launch.json`. It's an entry in the `package.json` in `keywords` array and
I believe that's a list that contains "tags" that users can search for when looking for an extension.

Here's an official description (search for "keywords"):
https://code.visualstudio.com/api/references/extension-manifest

Reply:
I know what the package.json keywords are for. There are few things that have to be in there and one is that if the extension uses the run & debug panel, whatever values are accepted in `type` in launch.json should also be package.json keywords. This is so that VSCode automatically runs the marketplace search when someone tries to run a debug session with that type and doesn't have the extension installed.

Reply:
Oh, I see. Only the following types are recognized:
![image](https://github.com/BroadcomMFD-DevOps/che-che4z-lsp-for-hlasm/assets/99467904/6022ea35-1469-4916-a511-9f1dc58b0ab5)

Reply:
okay, cool - since those look generic I guess they don't need to be in package.json. So the "ibm assembler" keyword is just there as an SEO keyword.

Reply:
Is the issue here that we shouldn't use "IBM"? If so I'm fine with removing it from both places

Reply:
> Is the issue here that we shouldn't use "IBM"? If so I'm fine with removing it from both places

I think that's the main question. I can remove it just to be on the safe side.

Reply:
Yes, this purely about those three letters...